.github/scripts/validateImmutableActionRefs.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+############################################
+#    Check for unsafe action references    #
+############################################
+
+GITHUB_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")/.." &>/dev/null && pwd)"
+readonly GITHUB_DIR
+
+source "$GITHUB_DIR/scripts/shellUtils.sh"
+
+title 'Checking for mutable action references...'
+
+ACTION_USAGES=''
+
+# Find yaml files - these can be either:
+# - workflows, which are always stored in .github/workflows, or
+YAML_FILES="$(find "$GITHUB_DIR/workflows" -type f \( -name "*.yml" -o -name "*.yaml" \))"
+# - action metadata files, which can be anywhere in the repo, but must be called action.yml or action.yaml
+YAML_FILES+=" $(find "$GITHUB_DIR/.." -type f \( -name "action.yml" -o -name "action.yaml" \))"
+
+# Find yaml files in the `.github` directory
+for FILE in $YAML_FILES; do
+    USES_LINES="$(grep 'uses:' "$FILE")"
```

Comment:
Style is a little different than other interpreted languages - prefer double quotes unless single quotes are required - see https://github.com/Expensify/Style-Guide/blob/main/shell.md#quoting

```suggestion
    USES_LINES="$(grep "uses:" "$FILE")"
```